lib/addons/argocd/index.ts

```diff
@@ -31,20 +32,22 @@ export interface ArgoCDAddOnProps extends HelmAddOnUserProps {
     /**
      * If provided, the addon will bootstrap the app or apps in the provided repository.
      * In general, the repo is expected to have the app of apps, which can enable to bootstrap all workloads,
-     * after the infrastructure and team provisioning is complete. 
+     * after the infrastructure and team provisioning is complete.
+     * When GitOps mode is enabled via `ArgoGitOpsFactory` for deploying the AddOns, this bootstrap
+     * repository will be used for provisioning all `HelmAddOn` based AddOns.
      */
     bootstrapRepo?: spi.ApplicationRepository;
```

Comment:
I also was thinking about an approach for non helm add-ons (just k8s manifests that we could deploy with some templating/kustomize) that could also be gitops driven. This is for later though, but let me know if you saw any customers asking for it.

Reply:
Good callout. Will definitely keep eye on this. Going through the current blueprint addon list, these are the only two not using HelmAddOn: (exclude those EKS managed addons)
- SSM Addon
- XRay Addon

Reply:
Yeah, I hope it will be less and less. Such add-ons may be, however, custom. SSM and XRay standalone are less of a concern for me atm. XRay support is available through AppMesh add-on (enableTracing).